_posts/2019-06-27-list-strategy.md

```diff
@@ -0,0 +1,126 @@
+- You might even completely forget about strategies configurations if the default behavior satisfies your needs.
+
+### Example #2
+This is how my [first example](#example-1) will look like now:
+
+``` 
+$ bazel build
+ --spawn_strategy=remote,local
+ ... 
+```
+
+And if I want a custom strategy for my Java rules I can still add `--strategy=Javac=`:
+
+```
+$ bazel build
+ --spawn_strategy=remote,local
+ --strategy=Javac=worker
+ ...
+```
+
+In this case, Java rules will be executed using a persistent *worker*, everything else will be executed *remotely* or *locally* (if remote is not possible).
+Isn't it much shorter and simpler?
+
+## How to use it
+
+- Don't set any flags and Bazel will try to do the best automatically: Use remote execution if it's available, otherwise persistent workers, otherwise sandboxed execution, otherwise non-sandboxed execution.
+- I want the best sandboxed build on my Linux machine and no automatic fallback to symlink-only sandboxing, non-sandboxed execution and no persistent workers: `--spawn_strategy=linux-sandbox`.
+- I want persistent workers for actions that support it, but otherwise only sandboxed execution: `--spawn_strategy=worker,sandboxed`.
+- I want to ensure everything runs remotely: `--spawn_strategy=remote`. Be careful, this configuration will cause a build failure in case bazel finds any action that cannot be executed remotely.
+
+### How to migrate
+You would know that you need to migrate if you see the following error:
+ 
+```
+ERROR: No usable spawn strategy found for spawn with mnemonic %Mnemonic%. Are your --spawn_strategy or --strategy flags too strict?
+```
+
+If that is the case or you want to migrate proactively, see below some ideas on how to fix it.
+
+- Do you use the strategy selection flags `--strategy` or `--spawn_strategy` or `--strategy_regexp`?  
+You'll want to revisit the values you set them to. If you get an error like above, consider adding a fallback strategy to your settings. For example, `--spawn_strategy=sandboxed,local` instead of `--spawn_strategy=sandboxed`.
+
+- Do you currently patch your Bazel version to remove fallback to non-sandboxed execution?  
+This should no longer be necessary then! 
+For example, something like `--spawn_strategy=linux-sandbox` will never fallback to `local` execution anymore.
+
+- Do you execute your build or part of it remotely using `--strategy=remote` and/or `--spawn_strategy=remote`?  
+Now you do not need to specify those strategies anymore - consider removing those `strategy` flags completely! In this case bazel will pickup the first available strategy from the default list, which is `remote`, unless you want to specifically forbid any non-remote executions or to configure custom fallback plan (see next advice).
+
+- If you were using custom fallback strategy for the remote execution, e.g. `--remote_local_fallback_strategy=worker`, just add a fallback strategy to the list, for example: `--spawn_strategy=remote,worker`. 
```

Comment:
I believe the advice should be to not set any `--remote_local_fallback_strategy` and `--spawn_strategy` flags at all instead.